src/main/java/org/example/calculator/model/CalculatorStringUtils.java

```diff
@@ -0,0 +1,32 @@
+package org.example.calculator.model;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CalculatorStringUtils {
+
+    public static final int FIRST_GROUP_OF_REGEX = 1;
+    static final String DEFAULT_DELIMITER = ",|:";
+    static final String CUSTOM_DELIMITER_REGEX = "//(.)\n";
+    static final String CUSTOM_OPERANDS_REGEX = "//.\n(.+)";
+
+    private CalculatorStringUtils() {
+
+    }
+
+    public static String extractDelimiterWithRegex(String string, String regex, String defaultDelimiter) {
+        Matcher matcher = Pattern.compile(regex).matcher(string);
+        if (matcher.find()) {
+            return matcher.group(FIRST_GROUP_OF_REGEX);
+        }
+        return defaultDelimiter;
+    }
+
+    public static String extractOperandStringWithRegex(String string, String regex) {
+        Matcher m = Pattern.compile(regex).matcher(string);
+        if (m.find()) {
+            return m.group(FIRST_GROUP_OF_REGEX);
+        }
+        return string;
+    }
```

Comment:
제가 조금 헷갈리게 코멘트를 남겨둔 것 같네요. 😅
아래와 같이 상수를 해당 클래스에 옮겨서 내부에서 직접 사용하도록 변경하여 `문자열 계산기` 도메인에 fit한 도메인 객체를 만들면 좋을 것 같다는 의미였어요. 😁

```suggestion
    public static String extractDelimiterWithRegex(String string) {
        Matcher matcher = Pattern.compile(CUSTOM_DELIMITER_REGEX).matcher(string);
        if (matcher.find()) {
            return matcher.group(FIRST_GROUP_OF_REGEX);
        }
        return DEFAULT_DELIMITER;
    }

    public static String extractOperandStringWithRegex(String string) {
        Matcher m = Pattern.compile(CUSTOM_OPERANDS_REGEX).matcher(string);
        if (m.find()) {
            return m.group(FIRST_GROUP_OF_REGEX);
        }
        return string;
    }
    }
```